airtable/plugin.py

```diff
@@ -115,58 +146,59 @@ def airtable_record_list(base_id: str, table_id_or_name: str) -> list[AirtableRe
 
 
 def airtable_record_create(
-    base_id: str, table_id_or_name: str, fields: dict[str, Any]
+    airtable_base_id: AirtableBaseID, airtable_table_id: AirtableTableID, fields: dict[str, Any]
 ) -> AirtableRecord:
     """
     Create a record using the Airtable `create records` API call with a POST.
-    base_id must be an ID and not a name.
     """
+    base_id = airtable_base_id.id
+    table_id = airtable_table_id.id
     with httpx.Client(
         transport=AugmentedTransport(actions_v0.authenticated_request_airtable)
     ) as client:
         response = client.post(
-            f"https://api.airtable.com/v0/{base_id}/{table_id_or_name}",
+            f"https://api.airtable.com/v0/{base_id}/{table_id}",
             json={"fields": fields},
         )
         if response.status_code != httpx.codes.OK:
             raise RuntimeError(
                 _resolve_error_message(
                     client,
                     base_id,
-                    table_id_or_name,
+                    table_id,
                     response.status_code,
                     response.text,
                 )
             )
         data = response.json()
     return AirtableRecord(
-        id=data["id"],
+        airtable_record_id=AirtableRecordID(data["id"]),
         created_time=datetime.fromisoformat(data["createdTime"]),
         fields=data["fields"],
     )
 
 
 def airtable_record_update_patch(
-    base_id: str, table_id_or_name: str, record_id: str, fields: dict[str, Any]
+    airtable_base_id: AirtableBaseID, airtable_table_id: AirtableTableID, airtable_record_id: AirtableRecordID, fields: dict[str, Any], typecast: bool = True
 ) -> None:
     """
     Update a record using the Airtable `update record` API call with a PATCH.
```

Comment:
Do we need to describe what `typecast` does in the docstring?

Reply:
yes, that should be added to the doc string